pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1141,13 +1141,16 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(clusterName string, service *v1.
 	var monitorIDs []string
 	for _, listener := range listenerList {
 		pool, err := getPoolByListenerID(lbaas.network, loadbalancer.ID, listener.ID)
-		if err != nil && err != ErrNotFound {
-			return fmt.Errorf("Error getting pool for listener %s: %v", listener.ID, err)
-		}
-		poolIDs = append(poolIDs, pool.ID)
-		// If create-monitor of cloud-config is false, pool has not monitor.
-		if pool.MonitorID != "" {
-			monitorIDs = append(monitorIDs, pool.MonitorID)
+		if err != nil {
+			if err != ErrNotFound {
+				return fmt.Errorf("Error getting pool for listener %s: %v", listener.ID, err)
+			}
+		} else {
```

Comment:
Thanks for catching and fixing this 
What about making it more clear, like ` if pool  != nil {...}`  and keep consistent with some other places doing the same check `err != nil && err != ErrNotFound` in this file